File: movie_finder/lib/data/model/movies/movie.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'movie.freezed.dart';
part 'movie.g.dart';

@freezed
class Movie with _$Movie {
  const factory Movie(
      {int? id,
      String? title,
      @JsonKey(name: 'poster_path') String? posterPath,
      String? overview,
      @JsonKey(name: 'release_date') String? releaseDate}) = _Movie;

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);
}

File: jobs/Frontend/movie-search/src/types/Movie.ts
type Movie = {
    id: number;
    title: string;
    tagline: string;
    overview: string;
    release_date: string;
    runtime: number;
    backdrop_path: string;
    vote_average: number;
    vote_count: number;
    genres: [{
        id: number,
        name: string
    }],
}

export default Movie;